integration-tests/awsguard.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integrationtests
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"regexp"
+
+	"github.com/pkg/errors"
+
+	ptesting "github.com/pulumi/pulumi/pkg/testing"
+)
+
+// Regex used to verify that policy names are reasonable.
+var ruleNameRE = regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9_]{1,100}$")
+
+// awsGuardSettings contains the configuration specific to the version of the AWS Guard
+// policy pack to be used for the integration test. This is how any specific configuration
+// values can be used for the test. (e.g. disabling or configuring rules.)
+type awsGuardSettings struct {
+	// Enforcement level to use for all policies. ("" will default to "mandatory".)
+	defaultEnforcementLevel string
+
+	// Specific rules to disable. (Will set their individual enforcement levels to "disabled".)
+	disableRules []string
```

Comment:
Consider naming this `disabledPolicies` rather than `disableRules` because "policy" is the terminology we use, not "rule".

Reply:
Alternatively, this could be a `map[string]string` (map of "policy key" to "enforcement level") rather than an array. Then, instead of a test having to disable specific policies, it could just set `defaultEnforcementLevel` to `"disabled"` (to disable all policies) and then enable only the policies it cares about testing.

Reply:
Both of those are good suggestions. I'll just rename the value to `disablePolicies` since that's the simpler thing to start with.